BlueShift-iOS-SDK/BlueShiftAppDelegate.m

```diff
@@ -165,8 +165,23 @@ - (BOOL)handleRemoteNotificationOnLaunchWithLaunchOptions:(NSDictionary *)launch
 - (void) registerForRemoteNotification:(NSData *)deviceToken {
     NSString *deviceTokenString = [[deviceToken description] stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"<>"]];
     deviceTokenString = [deviceTokenString stringByReplacingOccurrencesOfString:@" " withString:@""];
-    [BlueShift sharedInstance].deviceToken = deviceTokenString;
     [BlueShiftDeviceData currentDeviceData].deviceToken = deviceTokenString;
+    NSString *previousDeviceToken = [[BlueShift sharedInstance] getDeviceToken];
+    if (previousDeviceToken && deviceTokenString) {
+        if(![previousDeviceToken isEqualToString:deviceTokenString]) {
+            [self placeIdentifyCall];
+        }
+    } else if (deviceTokenString) {
+        [self placeIdentifyCall];
+    }
+}
+
+- (void)placeIdentifyCall {
+    [[BlueShift sharedInstance] setDeviceToken];
+    NSString *email = [BlueShiftUserInfo sharedInstance].email;
+    if (email && ![email isEqualToString:@""]) {
```

Comment:
lets fire an identify call even when the email is not set.
Non logged in users won't have email set, but we still need the identify call to be fired.